File: mycollab-ui/src/main/java/com/esofthead/mycollab/vaadin/ui/MyCollabSession.java
/**
 * This file is part of mycollab-ui.
 *
 * mycollab-ui is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-ui is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-ui.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.vaadin.ui;

import com.esofthead.mycollab.common.SessionIdGenerator;
import com.esofthead.mycollab.core.SessionExpireException;
import com.esofthead.mycollab.vaadin.MyCollabUI;
import com.vaadin.ui.UI;

/**
 * @author MyCollab Ltd.
 * @since 3.0
 */
public class MyCollabSession {

    public static final String EVENT_BUS_VAL = "eventBusVal";

    public static final String CURRENT_PROJECT = "project";

    public static final String PROJECT_MEMBER = "project_member";

    public static final String USER_TIMEZONE = "USER_TIMEZONE";

    public static final String CURRENT_MODULE = "currentModule";

    public static final String CONTROLLER_REGISTRY = "CONTROLLER_REGISTRY";

    public static final String HISTORY_VAL = "historyVal";

    public static final String PRESENTER_VAL = "presenterMap";

    public static final String VIEW_MANAGER_VAL = "viewMap";

    private MyCollabSession() {
    }

    /**
     * @param key
     * @param value
     */
    public static void putVariable(String key, Object value) {
        try {
            ((MyCollabUI) UI.getCurrent()).setAttribute(key, value);
        } catch (Exception e) {
            throw new SessionExpireException("Expire Exception");
        }
    }

    /**
     * @param key
     */
    public static void removeVariable(String key) {
        try {
            ((MyCollabUI) UI.getCurrent()).setAttribute(key, null);
        } catch (Exception e) {
            throw new SessionExpireException("Expire Exception");
        }
    }

    /**
     * @param key
     * @return
     */
    public static Object getVariable(String key) {
        try {
            return ((MyCollabUI) UI.getCurrent()).getAttribute(key);
        } catch (Exception e) {
            throw new SessionExpireException("Expire Exception");
        }
    }

    public static String getSessionId() {
        return SessionIdGenerator.getSessionId();
    }
}

File: mycollab-ui/src/main/java/com/mycollab/vaadin/ui/MyCollabSession.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.vaadin.ui

import com.mycollab.core.SessionExpireException
import com.mycollab.vaadin.AppUI
import com.vaadin.server.VaadinSession
import com.vaadin.ui.UI

/**
 * @author MyCollab Ltd.
 * @since 3.0
 */
object MyCollabSession {

    const val USER_VAL = "userVal"

    const val EVENT_BUS_VAL = "eventBusVal"

    const val CURRENT_PROJECT = "project"

    const val PROJECT_MEMBER = "project_member"

    const val CURRENT_MODULE = "currentModule"

    const val CONTROLLER_REGISTRY = "CONTROLLER_REGISTRY"

    const val PRESENTER_VAL = "presenterMap"

    const val VIEW_MANAGER_VAL = "viewMap"

    @JvmStatic fun putSessionVariable(key: String, value: Any) = try {
        VaadinSession.getCurrent().setAttribute(key, value)
    } catch (e: Exception) {
        throw SessionExpireException("Expire Exception")
    }

    @JvmStatic fun getSessionVariable(key: String): Any? = try {
        VaadinSession.getCurrent().getAttribute(key)
    } catch (e: Exception) {
        throw SessionExpireException("Expire Exception")
    }

    @JvmStatic fun removeSessionVariable(key: String) = try {
        VaadinSession.getCurrent().setAttribute(key, null)
    } catch (e: Exception) {
        throw SessionExpireException("Expire Exception")
    }

    /**
     * @param key
     * @param value
     */
    @JvmStatic fun putCurrentUIVariable(key: String, value: Any?) = try {
        (UI.getCurrent() as AppUI).setAttribute(key, value)
    } catch (e: Exception) {
        throw SessionExpireException("Expire Exception")
    }

    /**
     * @param key
     */
    @JvmStatic fun removeCurrentUIVariable(key: String) = try {
        (UI.getCurrent() as AppUI).setAttribute(key, null)
    } catch (e: Exception) {
        throw SessionExpireException("Expire Exception")
    }

    /**
     * @param key
     * @return
     */
    @JvmStatic fun getCurrentUIVariable(key: String): Any? = try {
        (UI.getCurrent() as AppUI).getAttribute(key)
    } catch (e: Exception) {
        throw SessionExpireException("Expire Exception")
    }
}